editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -432,9 +425,25 @@ impl Fsm for SelectToolFsmState {
 				// Update pivot
 				tool_data.pivot.update_pivot(document, &mut overlay_context);
 
-				// Update dragging box
+				// Check if the tool is in Selection Box mode
 				if matches!(self, Self::DrawingBox { .. }) {
-					overlay_context.quad(Quad::from_box([tool_data.drag_start, tool_data.drag_current]));
+					// get the updated box bounds
+					let quad = Quad::from_box([tool_data.drag_start, tool_data.drag_current]);
+
+					// Outline layers to be selected
+					for layer in document.intersect_quad(quad, &document.network) {
+						overlay_context.outline(document.metadata().layer_outline(layer), document.metadata().transform_to_viewport(layer));
+					}
+
+					// Update the selection box
+					overlay_context.quad(quad);
+				} else {
+					// Get the layer the user is hovering over
+					let click = document.click(input.mouse.position, &document.network);
+					let not_selected_click = click.filter(|&hovered_layer| !document.selected_nodes.selected_layers_contains(hovered_layer, document.metadata()));
+					if let Some(layer) = not_selected_click {
+						overlay_context.outline(document.metadata().layer_outline(layer), document.metadata().transform_to_viewport(layer));
```

Comment:
To avoid multiple `overlay_context.outline(...)` in this if .... else .... brace, a mutable Vec<layer_identifier> outside this if/else brace might be useful.

Reply:
@Tiger3018 hope the changes are what you require. Thanks for replying

Reply:
Sorry I wasn't able to code review this until now. @shyamjayakannan please revert the change requested by @Tiger3018 because it's less efficient and adds complexity than the good solution you had already chosen. Thanks.

Reply:
This code change is just fit to my suggestion earlier. However, towards the solution of `let mut Vec`, a simple `let xx = if` can also do the same thing while keeping the efficiency and simplicity. See [this answer](https://stackoverflow.com/a/59917403).

The second situation can be wrapped by `vec![]`. Because it should only have one layer result in the near future, I don't have a idea if this could introduce code latency @Keavon , since I have encountered multiple codes in Graphite like this.

Reply:
@Tiger3018 the linked answer only works if all branches give the same type (which here it wouldn't easily). Allocating on the heap does have some cost (although not particularly significant).

Reply:
@Keavon I have reverted the changes as requested. Thanks for the reply